### PR TITLE
Set up router and global context

### PR DESCRIPTION
**Pull request by vitalii-bulyzhyn**
This PR is mainly includes refactored code, there is no much new functionality. Main changes:
- router.ts - file where all the routing happening
- pages folder - contains all the ready to use pages
- context folder - contains context components (so for only lang)

Few more implementation details:
- now lang will be passed through Context API of React, so you can access it in any component with useContext hook
- for the routing there is a default react-router-dom

### CODE REVIEW

File: frontend/src/router.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import Home from "./pages/Home";
+import CanOfferShelter from "./pages/Signup/CanOfferShelter";
+import NeedShelter from "./pages/Signup/NeedShelter";
+
+export default function Router() {
+    return (
+        <BrowserRouter>
+            <Routes>
+                {/* Unprotexted routes */}
```
Comment (rmadsen) [inline frontend/src/router.tsx]: I think you mean Unprotected

File: frontend/src/router.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import About from "./pages/About";
+
+import Home from "./pages/Home";
+import CanOfferShelter from "./pages/Signup/CanOfferShelter";
+import NeedShelter from "./pages/Signup/NeedShelter";
+
+export default function Router() {
+    return (
+        <BrowserRouter>
+            <Routes>
```
Comment (rmadsen) [inline frontend/src/router.tsx:15]: I don't know much about React Router: I'm wondering could we just put the <Header> here so it shows up on every page?

File: frontend/src/context/lang.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { createContext } from 'react';
+
+export const LangContext = createContext([] as  any);
```
Comment (rmadsen) [inline frontend/src/context/lang.ts]: Could we type this? `createContext<[SupportedLocale, func] | []>([])`  (I'm not sure if that's exactly right...)

File: frontend/src/components/Header.tsx [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
 import { HamburgerIcon, CloseIcon } from '@chakra-ui/icons';
 import { Trans } from '@lingui/macro';
 import { SupportedLocale, locales } from '../i18n';
 import { SignInSignOutButton } from "./SignInSignOutButton";
+import { LangContext } from "../context/lang"
 
-export default function Header({ lang, onLangChange }: {lang: SupportedLocale, onLangChange: (v: SupportedLocale) => void}) {
+export default function Header() {
   const { isOpen, onOpen, onClose } = useDisclosure();
+  const [lang, setVal] = useContext(LangContext);
```
Comment (rmadsen) [inline frontend/src/components/Header.tsx]: A better term might be setLang

**Review by rmadsen — COMMENTED** [review]
Awesome, thanks for setting this up! I added a few comments—feel free to merge without fixing if you want though.

File: frontend/src/router.tsx (hunk 1 above)
Comment (vitalii-bulyzhyn) [reply frontend/src/router.tsx]: Yeah, of course, forgot to turn on spellcheck

File: frontend/src/router.tsx (hunk 2 above)
Comment (vitalii-bulyzhyn) [reply frontend/src/router.tsx:15]: You mean Header component?

File: frontend/src/context/lang.ts (hunk 3 above)
Comment (vitalii-bulyzhyn) [reply frontend/src/context/lang.ts]: Will try that out

File: frontend/src/components/Header.tsx (hunk 4 above)
Comment (vitalii-bulyzhyn) [reply frontend/src/components/Header.tsx]: Of course, will change

**Pull request merged**